pkg/operator/ceph/object/objectstore.go

```diff
@@ -524,27 +530,30 @@ func createSystemUser(objContext *Context, namespace string) error {
 	return nil
 }
 
-func setMultisite(objContext *Context, store *cephv1.CephObjectStore, serviceIP string) error {
+func setMultisite(objContext *Context, store *cephv1.CephObjectStore, isCustomZoneEndpointListPresent bool) error {
 	logger.Debugf("setting multisite configuration for object-store %v", store.Name)
-	serviceEndpoint := fmt.Sprintf("http://%s:%d", serviceIP, store.Spec.Gateway.Port)
-	if store.Spec.Gateway.SecurePort != 0 {
-		serviceEndpoint = fmt.Sprintf("https://%s:%d", serviceIP, store.Spec.Gateway.SecurePort)
-	}
+	serviceEndpoint := objContext.Endpoint
 
 	if store.Spec.IsMultisite() {
-		zoneEndpointsList, err := getZoneEndpoints(objContext, serviceEndpoint)
-		if err != nil {
-			return err
-		}
-		zoneEndpointsList = append(zoneEndpointsList, serviceEndpoint)
+		if !isCustomZoneEndpointListPresent {
```

Comment:
Instead of passing this bool around, could you just check the condition directly here? Something like:
```suggestion
		if len(store.Spec.CustomEndpoints) > 0 {
```

Reply:
Good point. We could also add a method to the spec like `store.Spec.CustomEndpointsAreSet()`.